design/als-design.md

```diff
@@ -0,0 +1,68 @@
+# AccessLog Service (ALS) Support
+
+Status: Draft
+
+## Abstract
+
+This document proposed a design for supporting Envoy's AccessLog Service capabilities in Contour.
+
+## Background
+
+Envoy has the ability to send rich access logs to access logging sinks.
+With AccessLog Service (ALS) enabled, Envoy can send access log messages using gRPC streams to a gRPC access log service. 
+
+This document focuses on configuring access log sinks and enabling Envoy to send access log messages via Contour.
+
+## Goals
+- Enable users to specify ALS configuration
+- Enable users to send HTTP access logs for specific listeners 
+
+## Non Goals
+- TCP Access Log Support
+
+## High-Level Design
+Contour will add support for Envoy's HTTP Access Log Service.
+ALS information will be specified as a static cluster for Envoy and Contour controller will set `access_log` field for each HTTPConnectionManager filter.
+
+## Detailed Design
+To enable Envoy to send access logs to a sink; first a static cluster needs to be set for Envoy.
+There are two ways to add this configuration:
+- ExtensionService
+- Contour Configuration file
+
+### ExtensionService
+Contour administrator/operator can add an ExtensionService with ALS configuration details.
+As part of `contour bootstrap`; the ExtensionService used to configure a static ALS cluster for Envoy.
```

Comment:
To do this, the bootstrap command will need a k8s client to first query the api for the ExtensionService. 

To me, setting args (or a config file) is much more simpler for this use-case and the extra indirection of adding an ExtensionService doesn't help but make it more confusing.

Reply:
The good thing about the ExtensionService is that it allows users to tune the timeouts, which will almost certainly be useful if you are using a sidecar.

I think it will also end up really confusing if we end up with a bunch of different ways to configure extra services in Envoy. We are using ExtensionService for auth and rate limiting already, and will probably need to for tracing. Although it's an extra step, it feels to me like it fits better to have the ExtensionService in there.

Reply:
What if we ditch the `contour bootstrap` initContainer and have the Leader in Contour manage this configmap? This would mean that Contour needs to write a configmap RBAC permission in a namespace, but would allow for a single config + setup.